_posts/2019-12-29-working-with-interfaces-in-go.md

```diff
@@ -0,0 +1,140 @@
+---
+layout: post
+title: Working with Interfaces in Go
+categories: [golang]
+---
+
+[Interfaces](https://golang.org/doc/effective_go.html#interfaces_and_types) are my favorite feature in Go. An interface type represents a set of methods. Unlike most other languages, you don’t have to explicitly declare that a type *implements* an interface. A struct **`S`** implements the interface **`I`** implicitly if **`S`** defines the methods that **`I`** requires. 
+
+Writing good interfaces is difficult. It's easy to [pollute](https://rakyll.org/interface-pollution/) the "API" of a package by exposing broad or unnecessary interfaces. In this article, we'll explain the reasoning behind existing guidelines for interfaces and supplement them with examples from the standard library. We'll also observe other patterns used in the library that don't conform to these best practices.
```

Comment:
The last sentence of this paragraph is fine, but within the article it would be helpful to indicate when an example is one where the standard library doesn't conform to the guideline in context. `library` -> `standard library`?

Reply:
I removed this last sentence, I think it makes it easier to follow